### DISCUSSION TITLE
How to get the number of active user subscriptions for the current topic, not consumers

### DISCUSSION THREAD

**Discussion by cxcel** [nats-io/nats.go#1682] [General]
### Proposed change

none

### Use case

none

**Comment by piotrpio** [comment]
Not to mix up concepts - by subscription you mean core NATS subscription, right? Because [consumers](https://docs.nats.io/nats-concepts/jetstream/consumers) refer to JetStream, which is a persistence layer on top of NATS and a [subscription](https://docs.nats.io/nats-concepts/core-nats/pubsub) refers to a stateless pubsub communication pattern.

If you want to get a list of active subscriptions in your nats-server (with the subjects they listen on), you can make use of built-in monitoring endpoints. You have 2 options:
- use http monitoring (start the server with -m flag, e.g. `nats-server -m 8080`), which exposes monitoring endpoints under the given port. You can then use e.g. `/connz?subs=true` endpoint to view all subscriptions per connection. Alternatively, you can use the `/subsz?subs=true` endpoint which will expose all subscriptions across all connections/accounts as a single list. You can read more about monitoring endpoints [here](https://docs.nats.io/running-a-nats-service/nats_admin/monitoring).
- use built-in NATS monitoring endpoints and send NATS requests to the server to get essentially the same data. HTTP and NATS endpoints cover the same data. You can read more about it [here](https://docs.nats.io/running-a-nats-service/configuration/sys_accounts).

**Comment by niondir** [comment]
I have maybe the same question. But like to add some context. No intention to capture this thread if I'm asking for something different.

I have 3 Consumers for my Stream and use them to distribute messages across 3 worker services. Since I need to guarantee that the same topic (partition) is always handled by the same worker.
But nothing keeps another client from consuming messages from my consumers. So it would help to check if the consumer is "active" already. Getting the number of subscriptions/connections would help. But maybe there is another metric I can use from somewhere.

I also keep searching myself.

**Reply by jnmoyne** [reply]
It is something different. There is a new feature that will be part of version 2.11 that will do what you are asking about (it's called 'pinned' client of a consumer): only one client pulling from the consumer is the pinned one (at a time) and only that client gets delivered messages.
